feat(framework:skip) Update Docker Compose `complete` files for 1.13.0

Robert-Steiner opened the pull request:
<!--
Thank you for opening a pull request (PR)!

Please rename your PRs following this [format](https://flower.ai/docs/framework/contributor-tutorial-contribute-on-github.html#pr-title-format).
Contribution guidelines: https://github.com/adap/flower/blob/main/CONTRIBUTING.md
-->

## Issue

### Description

Updates the Docker Compose complete setup to incorporate the upcoming 1.13.0 changes.

### Related issues/PRs

<!--
Link issues and/or PRs that are related to this PR.

Example: Fixes #123. See also #456 and #789.
-->

## Proposal

### Explanation

<!--
Explain the changes and how they improve the issue described above.

Example: The variable `rnd` was renamed to `server_round` to improve readability.
-->

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### Any other comments?

<!--
Please be aware that it may take some time until the maintainers can review the PR.
Smaller PRs with good descriptions can be considered much more easily.

If you have an urgent request or question, please use the Flower Slack:

    https://flower.ai/join-slack/ (channel: #contributions)

Thank you for contributing to Flower!
-->

chongshenng commented on src/docker/complete/compose.yml:
```diff
@@ -4,13 +4,19 @@ services:
     image: flwr/superlink:${FLWR_VERSION:-1.12.0}
```
```suggestion
    image: flwr/superlink:${FLWR_VERSION:-1.13.0}
```

Robert-Steiner replied:
good point!

chongshenng commented on src/docker/complete/compose.yml:
```diff
@@ -4,13 +4,19 @@ services:
       context: ${PROJECT_DIR:-.}
       dockerfile_inline: |
-        FROM flwr/superexec:${FLWR_VERSION:-1.12.0}
+        FROM flwr/serverapp:${FLWR_VERSION:-1.12.0}
```
```suggestion
        FROM flwr/serverapp:${FLWR_VERSION:-1.13.0}
```

chongshenng reviewed (changes requested):
@Robert-Steiner Thanks for this. I think the only the Docker image tags needs to be changed to `nightly` since it doesn't work with `1.12` right now.

Robert-Steiner commented:
> @Robert-Steiner Thanks for this. I think the only the Docker image tags needs to be changed to `nightly` since it doesn't work with `1.12` right now.

I changed it to nightly, but we’ll need to update it to 1.13.0 before releasing 1.13.0, to make it part of the release tag. I used to change the version after the release, but this caused an issue with the link in the [Run Flower Quickstart Examples with Docker Compose](https://flower.ai/docs/framework/docker/run-quickstart-examples-docker-compose.html) doc, where the link points to the correct tag, but the default version is still set to the previous release:

```
curl https://raw.githubusercontent.com/adap/flower/refs/tags/v1.12.0/src/docker/complete/compose.yml \
    -o compose.yml
```

I’ve added a note in the release docs.

chongshenng reviewed (approved):
lgtm!

chongshenng commented:
> I changed it to nightly, but we’ll need to update it to 1.13.0 before releasing 1.13.0, to make it part of the release tag

@Robert-Steiner yes, makes sense!

tanertopal reviewed (approved):
lgtm!
